modules/layers/src/geojson-layer/geojson-layer.ts

```diff
@@ -50,10 +51,19 @@ import {createLayerPropsFromFeatures, createLayerPropsFromBinary} from './geojso
 
 /** All properties supported by GeoJsonLayer */
 export type GeoJsonLayerProps<DataT extends Feature = Feature> = _GeoJsonLayerProps<DataT> &
-  CompositeLayerProps<DataT>;
+  Omit<CompositeLayerProps<DataT>, 'data'>;
```

Comment:
Why do we need the `Omit<>` if we are passing in the datatype? Can this be fixed in the layer type inheritance chain, so that the DataType we are passing in makes its way to the CompositeLayer. Code with a lot of such "member type picking" usually makes me uncomfortable, it does not help readability, especially for new contributors.

Reply:
Actual type of property would be `{ data: Foo } & { data: Foo | Bar }, that typescript treats as `{ data: Foo }`.
So we have to _remove_ old, incompatible type of data that comes from `LayerProps` before adding "new", desired one.

And yes, i agree this is very, very fragile and unpretty solution and after rechecking how it actually works, it seems like good experiment (even if failed :) ...)

There is no solution now to "enchance" base layer prop types unfortunately.

Reply:
Well - it does seem to me that we are doing this wrong. We should not be redeclaring `data` prop in each sublayers props, so that we have to `Omit<>` it from the base layer's props. We should control the type through type parameters to `LayerProps`.

Ideally a sublayer should not need to know the PropTypes for all its superclass chain, 

```typescript
type GeoJSONLayerProps = CompositeLayerProps<GeoJSONTypes, Feature[]> & { ... unique GeoJSONLayer props... )
```